File: 547-NumberOfProvinces/547-NumberOfProvinces.py
# Last updated: 6/11/2026, 11:32:40 AM
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        nodes = len(isConnected)
        visited = set()
        comp = [0]
        def dfs(node):
            if(node not in visited):
                visited.add(node)
                for i in range(len(isConnected)):
                    if(i!=node and isConnected[node][i]==1):
                        dfs(i)


        for i in range(nodes):
            if(i not in visited):
                dfs(i)
                comp[0] += 1
                
        return comp[0]
        

File: 547-NumberOfProvinces/547-NumberOfProvinces.cpp
// Last updated: 6/11/2026, 11:32:34 AM
class Solution {
public:
    void dfs(int i, vector<int> &vis, vector<vector<int>>& isConnected){
        int n = isConnected.size();
        vis[i] = 1;
        for(int j=1; j<=n; j++){
            if(isConnected[i-1][j-1] == 1 && vis[j] == 0){
                vis[j] = 1;
                dfs(j, vis, isConnected);
            }
        }
    }
    int findCircleNum(vector<vector<int>>& isConnected) {
        int n = isConnected.size();
        vector<int> vis(n+1, 0);
        int tp = 0;
        for(int i=1; i<=n; i++){
            if(vis[i] == 0){
                tp++;
                dfs(i, vis, isConnected);
            }
        }
        return tp;
    }
};